### PR TITLE
FAI-539: Counterfactual: Refactor support for "partial" goals in Trusty UI

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.redhat.com/browse/FAI-539

Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](https://github.com/kiegroup/kogito-runtimes#contributing-to-kogito)
- [x] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [x] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>Pull Request</b>  
  Please add comment: <b>Jenkins retest this</b>
 
* <b>Quarkus LTS checks</b>  
  Please add comment: <b>Jenkins run LTS</b>

* <b>Native checks</b>  
  Please add comment: <b>Jenkins run native</b>

</details>

### CODE REVIEW

**Comment by manstis** [comment]
@kelvah This, I believe, now reflects Missy's final(?) design and hence can be merged; if you're happy with it.

File: ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.tsx [hunk 1]
```diff
@@ -35,6 +35,16 @@ const CounterfactualOutcomeSelection = (
...
+  const areAllDesiredOutcomeFloating = useCallback(() => {
+    const supportedGoalCount = editingGoals.filter(
+      goal => goal.role !== CFGoalRole.UNSUPPORTED
+    ).length;
+    return (
+      editingGoals.filter(goal => goal.role === CFGoalRole.FLOATING).length ===
+      supportedGoalCount
+    );
+  }, [editingGoals]);
+
```
Comment (kelvah) [inline ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.tsx]: this is fine but I think `every` could play nice here. like:
```
const areAllDesiredOutcomeFloating = useCallback(() => {
    return editingGoals
       .filter(goal => goal.role !== CFGoalRole.UNSUPPORTED)
       .every(goal => goal.role === CFGoalRole.FLOATING);
  }, [editingGoals]);
```

File: ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.tsx [hunk 2]
```diff
@@ -79,82 +90,102 @@ const CounterfactualOutcomeSelection = (
...
+        description="Specify desired counterfactual outcomes for one or more original decision outcomes."
         actions={[
-          <Button
-            key="confirm"
-            variant="primary"
-            onClick={handleApply}
-            isDisabled={!isDesiredOutcomeDefined()}
+          // See https://github.com/patternfly/patternfly-react/pull/5991.
+          // We will currently get a memory leak error. Fixed in @patternfly/react-core@4.135.6
+          <Tooltip
```
Comment (kelvah) [inline ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.tsx:103]: just wondering, should the tool-tip be displayed only when the button is disabled? do we need it when everything is O.K.?

File: ui-packages/packages/trusty/src/components/Molecules/CounterfactualOutcomeEdit/CounterfactualOutcomeEdit.scss [hunk 3]
```diff
@@ -1,3 +1,4 @@
-.counterfactual-outcome__child {
-  padding-left: var(--pf-global--spacer--lg);
+.counterfactual-outcome__floating {
+  padding-top: var(--pf-global--spacer--form-element);
```
Comment (kelvah) [inline ui-packages/packages/trusty/src/components/Molecules/CounterfactualOutcomeEdit/CounterfactualOutcomeEdit.scss:2]: :+1: :+1: :+1:

File: ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.scss [hunk 4]
```diff
@@ -1,20 +1,34 @@
...
 .counterfactual__outcomes-grid {
   width: 100%;
   display: inline-grid;
-  grid-template-columns: repeat(2, 1fr);
+  grid-template-columns: 1fr 2fr;
   grid-column-gap: var(--pf-global--spacer--lg);
 
+  border-top: var(--pf-global--palette--black-300);
+  border-top-style: solid;
+  border-top-width: 1px;
```
Comment (kelvah) [inline ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.scss]: If you prefer a shorter syntax you could set them all with the shorthand property:
`border-top: 1px solid var(--pf-global--palette--black-300);`

**Review by kelvah — APPROVED** [review]
Hi @manstis, LGTM. I've left some really minor observations.

File: ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.tsx (hunk 1 above)
Comment (manstis) [reply ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.tsx]: Great! I'd have done something similar in the land of Java. I'll make the change here.

File: ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.tsx (hunk 2 above)
Comment (manstis) [reply ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.tsx:103]: No we don't. I'll make the change :+1:

File: ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.scss (hunk 4 above)
Comment (manstis) [reply ui-packages/packages/trusty/src/components/Organisms/CounterfactualOutcomeSelection/CounterfactualOutcomeSelection.scss]: @kelvah Updates completed.

**Pull request merged**